### ISSUE TITLE
TSM causes Firefox 126.0.1 (64bit) to close immediately after opening Firefox

### ISSUE THREAD

**Issue by trumpy81** [sienori/Tab-Session-Manager#1342]
## Short description
When opening Firefox, it immediately closes. 

## Steps to reproduce
1. Start browser with clean profile
2. Install Tab Session Manager
3. Configure TSM Settings > Startup to 'Open Previous Session'
4. Open Firefox, it closes immediately

## Expected result
To open Firefox without immediately closing.

## Actual result
If Settings > Startup is set to 'Open Previous Session', Firefox closes immediately. If Settings > Startup is set to 'Open Startup Session' or 'Do Nothing', then Firefox opens and the previous tabs need to be opened from the TSM Toolbar Icon.

## Platform information
* Platform (OS): Windows 11 Pro (64bit)
* Version of browser: Firefox 126.0.1 (64bit)
* Version of Tab Session Manager: 7.0.0

## Additional context
I previously had this same issue back in 2022, see closed issue #1076 and the solution at that time was to set the Firefox home page using a physical URL instead of the default, this time however, that solution is not working.

**Comment by irasanen** [comment]
I think same as this:
https://github.com/sienori/Tab-Session-Manager/issues/1333

https://github.com/sienori/Tab-Session-Manager/issues/1333#issuecomment-2150428455
_"This issue will be fixed in version 7.0.1."_

**Comment by trumpy81** [comment]
Thanks for the info @irasanen.

I will wait for TSM 7.0.1 to drop and again, thanks for filling me in :)

**Comment by TeaganXXX** [comment]
I've also been having the same issue for several days, what do I do???   Help?!?

**Comment by TeaganXXX** [comment]
> ## Short description
> 
> When opening Firefox, it immediately closes.
> ## Steps to reproduce
> 
>     1. Start browser with clean profile
> 
>     2. Install Tab Session Manager
> 
>     3. Configure TSM Settings > Startup to 'Open Previous Session'
> 
>     4. Re-open Firefox, it closes immediately
> 
> 
> ## Expected result
> 
> To open Firefox without immediately closing.
> ## Actual result
> 
> If Settings > Startup is set to 'Open Previous Session', Firefox closes immediately. If Settings > Startup is set to 'Open Startup Session' or 'Do Nothing', then Firefox opens and the previous tabs need to be opened from the TSM Toolbar Icon.
> ## Platform information
> 
>     * Platform (OS): Windows 11 Pro (64bit)
> 
>     * Version of browser: Firefox 126.0.1 (64bit)
> 
>     * Version of Tab Session Manager: 7.0.0
> 
> 
> ## Additional context
> 
> I previously had this same issue back in 2022, see closed issue #1076 and the solution at that time was to set the Firefox home page using a physical URL instead of the default, this time however, that solution is not working.


I've also been having the same issue for several days, what do I do??? Help?!?

**Comment by trumpy81** [comment]
@TeaganXXX For now, the only thing you can do is go to TSM Settings > Startup and set it to 'Open Startup Session' or 'Do Nothing', then Firefox will at least open and the previous tabs can be manually opened from the TSM Toolbar Icon.

**Comment by TeaganXXX** [comment]
@trumpy81  Thank you, appreciate you replying. 👍🏼  I hope the developer fixes it as soon as possible. 

 Who needs a functioning browser anyway, right?   Seriously, I've never encountered anything quite like this before, especially with it dragging on for so long.

**Comment by trumpy81** [comment]
You obviously think that debugging computer code is an easy process, believe me it is not!

It can take a considerable amount of time and effort to find the root cause/s for any issues. The fact that Firefox was updated before this issue occurred tells me that something unexpected may have changed in Firefox.

I doubt that **sienori** could have predicted that and they are now faced with a lengthy, time consuming bug hunt. 

**sienori** has been one of the more responsive developers, so I am confident they will track down the bug and correct it just as soon as possible.

In the meantime, you need to exercise some patience and consider supporting **sienori** for the excellent work they have done (free of charge) thus far, without their efforts, neither of us would be here.

**Comment by sienori** [comment]
Sorry for the inconvenience. I have already fixed this issue and uploaded version 7.0.1 to the Add-on Store. Please wait a few days until it is published in the Add-on Store.

**Comment by trumpy81** [comment]
@sienori

Thank you very much. I will monitor the add-on store and download v7.0.1 as soon as it is available and thanks again for your excellent support! :thumbsup:

**Comment by EvVlF** [comment]
i think i have same issue.
Linux, Ff 126.0.1, x64

```
 [Parent 4277, Main Thread] WARNING: Failed to enumerate devices of org.freedesktop.UPower: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
: 'glib warning', file /usr/src/debug/firefox/firefox-126.0.1/toolkit/xre/nsSigHandlers.cpp:187

** (firefox:4277): WARNING **: 20:24:13.820: Failed to enumerate devices of org.freedesktop.UPower: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
```

when i start Ff with 'firefox -p', 'firefox -safe-mode' or with disabled tab session - all work.

**Comment by trumpy81** [comment]
Just a follow up. TSM v7.0.1 dropped for me this morning, I installed it and the issue with Firefox startup is gone.

Upon starting Firefox, all previous tabs were restored.

Thanks for your hard work @sienori it is very much appreciated :smiley:

**Comment by TeaganXXX** [comment]
> You obviously think that debugging computer code is an easy process, believe me it is not!
> 
> It can take a considerable amount of time and effort to find the root cause/s for any issues. The fact that Firefox was updated before this issue occurred tells me that something unexpected may have changed in Firefox.
> 
> I doubt that **sienori** could have predicted that and they are now faced with a lengthy, time consuming bug hunt.
> 
> **sienori** has been one of the more responsive developers, so I am confident they will track down the bug and correct it just as soon as possible.
> 
> In the meantime, you need to exercise some patience and consider supporting **sienori** for the excellent work they have done (free of charge) thus far, without their efforts, neither of us would be here.


@trumpy81 

You are in no position to claim whether or not I was “obviously” thinking anything, or make assumptions in regard to what I think and/or know about the process of debugging computers, programs, software, or codding.  I have a right to my opinion, and I stand by what I wrote, sarcasm and all.  

Regarding Mozilla Firefox's release of Version 126.0.1 on May 28, 2024, I personally began experiencing issues with the Tab-Session-Manager Extension or Add-On four to five days later. To keep this concise, my concern lies not with the developer @sienori, but with the lack of transparency from Mozilla Firefox toward extension developers whose creations are utilized by thousands, if not hundreds of thousands, of users. The evident lack of communication exacerbates this issue.  

Surely, this isn't the first time an incident like this has occurred with an extension, right? Consider this: if Mozilla Firefox were a for-profit subsidiary of the Mozilla Foundation, and one day, due to an extension issue, hundreds of thousands of users found themselves unable to keep their browsers open, would it have taken several days to address the problem?  

Developers do receive release notes before Firefox rolls out a new version.  However, it's increasingly apparent that they (Firefox) aren't providing developers with all the necessary information, putting them in a challenging position.  

You might want to stop making assumptions of what others are thinking before you say anything.

**Comment by trumpy81** [comment]
I made no assumptions, the evidence is clearly in your writings and that is what I based my observation upon.

If you intended to direct your criticism at Mozilla, then perhaps you should have mentioned that in your post?

The issue with add-ons/plugins is an ongoing issue and it has caused some companies to drop add-ons/plugins altogether, to the detriment of their users.

Unfortunately, I see no easy solution for the problems you outlined and I suspect there will be many such incidents in the future.

**Comment by EvVlF** [comment]
> @EvVlF Instead of posting blindly, perhaps you should have read this thread, the last two posts at least. If you had, then you may know that the issue has already been identified and fixed.
> 
> We are now, simply waiting for the updated version (v7.0.1) of TSM to become available in the Firefox add-on store.

Instead of posting "meh-meh-meh, im software developer princess, 'im so sick of these plebeian users" you will see - for example, this may not only apply to this ticket - perhaps the message will contain new information for developers. You'll also see that nowhere did I demand that anything be fixed immediately. 
As I see it - you've already annoyed at least two people here. Maybe you should reclassify from developer to zookeeper, since your soft skills are shit.

**Comment by TeaganXXX** [comment]
@trumpy81  

Yes, you actually did make an assumption, and I also did direct criticism at Mozilla.  If you think "the evidence is clearly in your writings" then you must also judge a book by its cover.  🙄  The developer @sienori, once informed of an issue, they executed a temporary fix, which was followed by an update of v7.0.1  

Now, the update was sent out, **but** it took Mozilla Firefox nearly 48 hours before the update was made available to the public.  Which brings me to what you wrote, "Unfortunately, I see no easy solution for the problems you outlined" ... If you really believe that, then I suggest you remain seated in the bleachers with the peanut gallery.  

I am a Firefox browser user and have been for years.  Mozilla Firefox known for its strong privacy features, commitment to open standards, and community-driven development.  But from my own observation, it's always lacked in 'communication'.  Until that changes, these "easy solutions" as you described it will continue, and Mozilla Firefox will remain at number 4 of most popular web browsers.  

     . , , enjoy seating in bleachers @trumpy81 with the peanut gallery.
